src/shared_modules/utils/stdFileSystemHelper.hpp

```diff
@@ -0,0 +1,113 @@
+/*
+ * Wazuh shared modules utils
+ * Copyright (C) 2015, Wazuh Inc.
+ * July 23, 2023.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef _STD_FILESYSTEM_HELPER_HPP
+#define _STD_FILESYSTEM_HELPER_HPP
+
+#include "stringHelper.h"
+#include <array>
+#include <filesystem>
+#include <string>
+#include <vector>
+
+
+namespace Utils
+{
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-function"
+
+    static void expandAbsolutePath(const std::string& path, std::vector<std::string>& output)
```

Comment:
Maybe, a new parameter with the type of file (REGULAR, DIRECTORY, etc) could be added to reduce useless result entries. Which could result in a performance improvement for later stages.

Reply:
the idea is to forget about this from the caller.